packages/generator/templates/app/app/auth/mutations/forgotPassword.ts

```diff
@@ -0,0 +1,51 @@
+import { SessionContext } from "blitz"
+import db from "db"
+import { forgotPasswordMailer } from "mailers/forgotPasswordMailer"
+import { ForgotPasswordInput, ForgotPasswordInputType } from "../validations"
+import {
+  generateToken,
+  hashToken,
+  RESET_PASSSWORD_TOKEN_EXPIRATION_IN_HOURS,
+} from "app/auth/auth-utils"
+
+export default async function forgotPassword(
+  input: ForgotPasswordInputType,
+  _ctx: { session?: SessionContext } = {}
+) {
+  const { email } = ForgotPasswordInput.parse(input)
+
+  // 1. Get the user
+  const user = await db.user.findOne({ where: { email: email.toLowerCase() } })
+
+  // 2. Generate the token and expiration date.
+  // We use encodeURIComponent(token) since it will be used in the URL
+  const token = generateToken()
```

Comment:
suggestion: Consider putting all this computation in the `if(user)` block, no real need to calculate hashes etc if they will not be used.